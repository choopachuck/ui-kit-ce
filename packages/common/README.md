# @v-uik/common

Общие объекты, интерфейсы и т.п.

- [ElementSize](#elementsize)
- [Direction](#direction)

## ElementSize

Перечисляет размеры элементов.

- **sm** - маленький размер
- **md** - стандартный размер
- **lg** - большой размер

Пример использования:

```javascript
import * as React from 'react'
import { clsx } from '@v-uik/theme'
import { ElementSize } from '@v-uik/common'

export const Component = (props) => {
  const className = clsx({
    small: props.size === ElementSize.SM,
    large: props.size === ElementSize.LG,
    medium: props.size === ElementSize.MD,
  })

  return <div {...props} className={className} />
}
```

## Direction

Направление (раскладка) элемента.

- **horizontal** - горизонтальная раскладка элемента, слева направо
- **vertical** - вертикальная раскладка элемента, сверху вниз

Пример использования:

```javascript
import * as React from 'react'
import { clsx } from '@v-uik/theme'
import { Direction } from '@v-uik/common'

export const Component = (props) => {
  const className = clsx({
    horizontal: props.dir === Direction.horizontal,
    vertical: props.dir === Direction.vertical,
  })

  return <div {...props} className={className} />
}
```
