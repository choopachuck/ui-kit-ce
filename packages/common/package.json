{
  "name": "@v-uik/common",
  "version": "1.10.0",
  "description": "",
  "main": "dist/cjs/index.js",
  "typings": "dist/esm/index.d.ts",
  "files": [
    "/dist/",
    "package.json",
    "README.md"
  ],
  "scripts": {
    "clean": "rm -rf node_modules dist",
    "build": "yarn build:cjs && yarn build:esm",
    "build:cjs": "tsc -d --outDir ./dist/cjs --project tsconfig.cjs.json",
    "build:esm": "tsc -d --outDir ./dist/esm --project tsconfig.esm.json"
  },
  "dependencies": {
    "@v-uik/hooks": "^1.10.0",
    "@v-uik/theme": "^1.10.0",
    "@v-uik/utils": "^1.10.0"
  },
  "author": "UFSUI",
  "license": "ISC",
  "sideEffects": false,
  "module": "dist/esm/index.js"
}
