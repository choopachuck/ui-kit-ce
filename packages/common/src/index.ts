export * from './Direction'
export * from './ElementSize'
export * from './TrapFocus'
export * from './breakpoints'
export * from './ClearButton'
