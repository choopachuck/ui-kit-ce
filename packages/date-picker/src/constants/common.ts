export const DatePickerView = {
  day: 'day',
  month: 'month',
  year: 'year',
} as const

export type DatePickerViewType = keyof typeof DatePickerView

export const defaultViews: DatePickerViewType[] = ['day', 'month', 'year']

export enum DateValidationError {
  invalidDate = 'invalidDate',
  notAllowedDate = 'notAllowedDate',
  isBeforeStartDate = 'isBeforeStartDate',
  isAfterEndDate = 'isAfterEndDate',
}

export type DateValidationErrorMessages = {
  [key in DateValidationError]: string
}

export const defaultValidationErrorMessages: DateValidationErrorMessages = {
  [DateValidationError.invalidDate]: 'Некорректная дата',
  [DateValidationError.notAllowedDate]: 'Дата недоступна для выбора',
  [DateValidationError.isBeforeStartDate]: 'Дата конца раньше даты начала',
  [DateValidationError.isAfterEndDate]: 'Дата начала позже даты конца',
}

export enum TimeValidationError {
  invalidTime = 'invalidTime',
  notAllowedTime = 'notAllowedTime',
  isBeforeStartTime = 'isBeforeStartTime',
  isAfterEndTime = 'isAfterEndTime',
}

export type TimeValidationErrorMessages = {
  [key in TimeValidationError]: string
}
