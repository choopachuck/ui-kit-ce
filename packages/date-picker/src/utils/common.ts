export const splitArrayIntoChunks = <T = unknown>(
  array: Array<T>,
  chunkSize = 0
): Array<Array<T>> => {
  const result: Array<Array<T>> = []

  if (chunkSize > 0) {
    while (array.length) {
      result.push(array.splice(0, chunkSize))
    }
  }

  return result
}

export const noop = (): void => {} // eslint-disable-line @typescript-eslint/no-empty-function
