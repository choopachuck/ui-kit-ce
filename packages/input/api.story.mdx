import { Meta, ArgsTable } from '@storybook/addon-docs'
import { Input } from '@v-uik/input'
import { InputHelperText } from '@v-uik/input-helper-text'
import { InputLabel } from '@v-uik/input-label'
import { createTitle, COMPONENTS } from '@v-uik/showroom/config'
import {
  ClassesDummy,
  InputLabelClassesDummy,
  InputHelperTextClassesDummy,
} from './examples'

<Meta
  title={createTitle([COMPONENTS.inputFields, 'Input', 'API'])}
  component={Input}
  parameters={{
    viewMode: 'docs',
    previewTabs: {
      canvas: { hidden: true },
    },
  }}
/>

# Input API

<ArgsTable of={Input} />

# Input classes

<ArgsTable of={ClassesDummy} />

# InputLabel API

<ArgsTable of={InputLabel} />

# InputLabel classes

<ArgsTable of={InputLabelClassesDummy} />

# InputHelperText API

<ArgsTable of={InputHelperText} />

# InputHelperText classes

<ArgsTable of={InputHelperTextClassesDummy} />
