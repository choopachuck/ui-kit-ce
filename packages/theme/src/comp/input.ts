import { Theme } from '../interface'

// eslint-disable-next-line @typescript-eslint/explicit-module-boundary-types
export const createInput = (theme: Theme, compatibilityMode?: boolean) => ({
  colorBackground: compatibilityMode
    ? theme.colourway.backgroundComponent
    : theme.sys.color.backgroundComponent,
  colorBackgroundDisabled: compatibilityMode
    ? theme.colourway.backgroundComponent
    : theme.sys.color.disabledLow,
  colorBorder: compatibilityMode
    ? theme.colourway.mono05
    : theme.sys.color.neutralAlpha,
  colorBorderHover: compatibilityMode
    ? theme.colourway.mono06
    : theme.sys.color.neutralBeta,
  colorBorderError: compatibilityMode
    ? theme.colourway.error
    : theme.sys.color.errorAlpha,
  colorBorderDisabled: compatibilityMode
    ? theme.colourway.disabledMajor
    : theme.sys.color.disabledMedium,
  colorText: compatibilityMode
    ? theme.colourway.onBackgroundHigh
    : theme.sys.color.onBackgroundHigh,
  colorTextDisabled: compatibilityMode
    ? theme.colourway.onBackgroundDisabled
    : theme.sys.color.disabledHigh,
  colorShadowFocus: compatibilityMode
    ? theme.colourway.focus
    : theme.sys.color.focus,
  placeholderColorText: compatibilityMode
    ? theme.colourway.onBackgroundLow
    : theme.sys.color.onBackgroundLow,
  placeholderColorTextDisabled: compatibilityMode
    ? theme.colourway.onBackgroundDisabled
    : theme.sys.color.disabledHigh,
  alertIconColorText: compatibilityMode
    ? theme.colourway.error
    : theme.sys.color.errorAlpha,
  prefixColorText: compatibilityMode
    ? theme.colourway.onBackgroundLow
    : theme.sys.color.onBackgroundLow,
  prefixColorTextDisabled: compatibilityMode
    ? theme.colourway.onBackgroundDisabled
    : theme.sys.color.disabledHigh,
  suffixColorText: compatibilityMode
    ? theme.colourway.secondary
    : theme.sys.color.secondaryAlpha,
  suffixColorTextDisabled: compatibilityMode
    ? theme.colourway.onBackgroundDisabled
    : theme.sys.color.disabledHigh,

  shapeBorderRadiusTopLeftMd: compatibilityMode
    ? theme.shape.borderRadius
    : theme.sys.shape.borderRadiusMd,
  shapeBorderRadiusTopRightMd: compatibilityMode
    ? theme.shape.borderRadius
    : theme.sys.shape.borderRadiusMd,
  shapeBorderRadiusBottomLeftMd: compatibilityMode
    ? theme.shape.borderRadius
    : theme.sys.shape.borderRadiusMd,
  shapeBorderRadiusBottomRightMd: compatibilityMode
    ? theme.shape.borderRadius
    : theme.sys.shape.borderRadiusMd,

  shapeBorderRadiusTopLeftSm: compatibilityMode
    ? theme.shape.borderRadius
    : theme.sys.shape.borderRadiusSm,
  shapeBorderRadiusTopRightSm: compatibilityMode
    ? theme.shape.borderRadius
    : theme.sys.shape.borderRadiusSm,
  shapeBorderRadiusBottomLeftSm: compatibilityMode
    ? theme.shape.borderRadius
    : theme.sys.shape.borderRadiusSm,
  shapeBorderRadiusBottomRightSm: compatibilityMode
    ? theme.shape.borderRadius
    : theme.sys.shape.borderRadiusSm,

  shapeBorderRadiusTopLeftLg: compatibilityMode
    ? theme.shape.borderRadius
    : theme.sys.shape.borderRadiusLg,
  shapeBorderRadiusTopRightLg: compatibilityMode
    ? theme.shape.borderRadius
    : theme.sys.shape.borderRadiusLg,
  shapeBorderRadiusBottomLeftLg: compatibilityMode
    ? theme.shape.borderRadius
    : theme.sys.shape.borderRadiusLg,
  shapeBorderRadiusBottomRightLg: compatibilityMode
    ? theme.shape.borderRadius
    : theme.sys.shape.borderRadiusLg,

  typographyFontFamily: compatibilityMode
    ? theme.typography.fontFamily.text
    : theme.sys.typography.bodyMd.fontFamily,
  typographyFontWeight: compatibilityMode
    ? theme.typography.fontWeight.regular
    : theme.sys.typography.bodyMd.fontWeight,
  typographyFontSize: compatibilityMode
    ? theme.typography.fontSize.body2
    : theme.sys.typography.bodyMd.fontSize,
  typographyLineHeight: compatibilityMode
    ? theme.typography.lineHeight.body2
    : theme.sys.typography.bodyMd.lineHeight,
  typographyLetterSpacing: compatibilityMode
    ? theme.typography.letterSpacing.body2
    : theme.sys.typography.bodyMd.letterSpacing,
})
