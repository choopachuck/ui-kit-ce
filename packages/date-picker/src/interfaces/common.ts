import { DatePickerViewType } from '../constants/common'
import { ButtonProps } from '@v-uik/button'
import { DayViewProps } from '../views/DayView'
import { MonthViewProps } from '../views/MonthView'
import { YearViewProps } from '../views/YearView'

export type ParsableDate<TDate = unknown> =
  | string
  | number
  | Date
  | null
  | undefined
  | TDate

export type CheckDateStateResult = {
  value: boolean
  preventNextCheck?: boolean
  position?: 'start' | 'end'
}

export type ExistedViews = {
  [key in DatePickerViewType]: boolean
}

export interface ForwardRefExoticComponentCommonFields<
  P = Record<string, never>
> {
  readonly $$typeof: symbol
  displayName?: string
  defaultProps?: Partial<P>
  propTypes?: React.WeakValidationMap<P>
}

export interface FunctionComponentCommonFields<P = Record<string, never>> {
  propTypes?: React.WeakValidationMap<P>
  contextTypes?: React.ValidationMap<unknown>
  defaultProps?: Partial<P>
  displayName?: string
}

/**
 * Тип для доп пропсов в календарь
 */
export type ExternalCalendarViewComponentsProps<TDate extends unknown> = {
  /**
   * Пропсы предыдщуей кнопки навигации
   */
  prevNavigationBarButtonProps: Omit<
    ButtonProps,
    'kind' | 'color' | 'className' | 'onClick' | 'children'
  >
  /**
   * Пропсы следующей кнопки навигации
   */
  nextNavigationBarButtonProps: Omit<
    ButtonProps,
    'kind' | 'color' | 'className' | 'onClick' | 'children'
  >
  /**
   * Пропсы кнопки месяца в панеле
   */
  monthBarButtonProps: Omit<
    ButtonProps,
    'kind' | 'color' | 'className' | 'onClick' | 'children'
  >
  /**
   * Пропсы кнопки год в панеле
   */
  yearBarButtonProps: Omit<
    ButtonProps,
    'kind' | 'color' | 'className' | 'onClick' | 'children'
  >
  /**
   * Пропсы для DayView
   */
  dayViewProps: Omit<
    DayViewProps<TDate>,
    | 'value'
    | 'displayedDate'
    | 'shouldDisableDate'
    | 'onChangeDisplayedDate'
    | 'onChange'
  >
  /**
   * Пропсы для MonthView
   */
  monthViewProps: Omit<
    MonthViewProps<TDate>,
    | 'value'
    | 'displayedDate'
    | 'shouldDisableDate'
    | 'onChangeDisplayedDate'
    | 'onChange'
  >
  /**
   * Пропсы для YearView
   */
  yearViewProps: Omit<
    YearViewProps<TDate>,
    | 'autoFocus'
    | 'value'
    | 'displayedDate'
    | 'shouldDisableDate'
    | 'minDate'
    | 'maxDate'
    | 'onChangeDisplayedDate'
    | 'onChange'
  >
}

export type ExternalCalendarViewComponentsPropsPartial<TDate extends unknown> =
  Partial<ExternalCalendarViewComponentsProps<TDate>>
