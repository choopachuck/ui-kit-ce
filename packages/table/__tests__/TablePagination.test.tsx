import * as React from 'react'
import {
  TablePaginationDirectionType,
  TablePagination,
  TablePaginationProps,
  PickerProps,
} from '../src'
import { fireEvent, render, screen } from '@testing-library/react'
import { ElementSizeType } from '@v-uik/common'
import { Input } from '@v-uik/input'

const Component = (props: Partial<TablePaginationProps>) => {
  const {
    currentPage: initialCurrentPage = 1,
    pageSize: initialPageSize = 10,
    ...rest
  } = props
  const [currentPage, setCurrentPage] = React.useState(initialCurrentPage)
  const [pageSize, setPageSize] = React.useState(initialPageSize)

  return (
    <TablePagination
      data-testid="pagination"
      currentPage={currentPage}
      pageSize={pageSize}
      totalCount={100}
      onChange={setCurrentPage}
      onPageSizeChange={setPageSize}
      {...rest}
    />
  )
}

const components = {
  PageSizer: ({ onChange, value }: PickerProps) => {
    return (
      <Input
        inputProps={{ 'aria-label': 'размер страницы' }}
        value={value}
        onChange={onChange}
      />
    )
  },
  PagePicker: ({ onChange, value }: PickerProps) => {
    return (
      <Input
        inputProps={{ 'aria-label': 'выбор страницы' }}
        value={value}
        onChange={onChange}
      />
    )
  },
}

it('render with correct props', () => {
  const { getByTestId } = render(<Component />)

  expect(getByTestId('pagination')).toMatchSnapshot()
})

it.each(['sm', 'md', 'lg'])('render pagination with size props', (size) => {
  const { getByTestId } = render(<Component size={size as ElementSizeType} />)

  expect(getByTestId('pagination').className).toMatch(size)
})

it.each(['start', 'end'])(
  'render pagination with direction props',
  (direction) => {
    const { getByTestId } = render(
      <Component direction={direction as TablePaginationDirectionType} />
    )

    expect(getByTestId('pagination').className).toMatch(direction)
  }
)

it('render without absoluteArrows', () => {
  const { getAllByRole, queryByRole } = render(
    <Component paginationType="simple" absoluteArrows={false} />
  )

  expect(getAllByRole('button')).toHaveLength(2)
  expect(getAllByRole('button')).not.toHaveLength(4)
  expect(
    queryByRole('button', { name: 'предыдущая страница' })
  ).toBeInTheDocument()
  expect(
    queryByRole('button', { name: 'следующая страница' })
  ).toBeInTheDocument()
  expect(
    queryByRole('button', { name: 'последняя страница' })
  ).not.toBeInTheDocument()
  expect(
    queryByRole('button', { name: 'последняя страница' })
  ).not.toBeInTheDocument()
})

it('render absolute arrows', () => {
  const { getAllByRole, queryByRole } = render(<Component absoluteArrows />)

  expect(getAllByRole('button')).toHaveLength(4)
  expect(getAllByRole('button')).not.toHaveLength(2)

  expect(
    queryByRole('button', { name: 'предыдущая страница' })
  ).toBeInTheDocument()
  expect(
    queryByRole('button', { name: 'следующая страница' })
  ).toBeInTheDocument()
  expect(
    queryByRole('button', { name: 'последняя страница' })
  ).toBeInTheDocument()
  expect(queryByRole('button', { name: 'первая страница' })).toBeInTheDocument()
})

it('render pageRangeText', () => {
  const { getByText } = render(
    <Component
      currentPage={2}
      totalCount={100}
      pageSize={25}
      pageRangeText={(min, max) => `${min} из ${max} листов`}
    />
  )

  expect(getByText('2 из 4 листов')).toBeInTheDocument()
})

it('render itemRangeText', () => {
  const { getByText } = render(
    <Component
      paginationType="advanced"
      currentPage={2}
      totalCount={100}
      pageSize={25}
    />
  )

  expect(getByText('26–50 из 100 элементов')).toBeInTheDocument()
})

it('divider direction', () => {
  const { getByTestId } = render(
    <Component
      test-id="pagination"
      paginationType="advanced"
      currentPage={2}
      totalCount={100}
      pageSize={25}
      divider="bottom"
    />
  )

  expect(getByTestId('pagination').className).toMatch(/dividerBottom/i)
})

it('currentPage=1 - change on page size dont trigger event onChange', () => {
  const onChange = jest.fn()

  const { getByRole } = render(
    <Component
      paginationType="advanced"
      components={components}
      onChange={onChange}
    />
  )

  fireEvent.change(getByRole('textbox', { name: 'размер страницы' }), {
    target: { value: 20 },
  })
  expect(onChange).toHaveBeenCalledTimes(0)
})

it('currentPage>1 - change on page size trigger event onChange', () => {
  const onChange = jest.fn()

  const { getByRole } = render(
    <Component
      paginationType="advanced"
      currentPage={2}
      components={components}
      onChange={onChange}
    />
  )

  fireEvent.change(getByRole('textbox', { name: 'размер страницы' }), {
    target: { value: 20 },
  })
  expect(onChange).toHaveBeenCalledTimes(1)
})

it('move to first page after change page size', async () => {
  const { getByRole, getByText, findByText } = render(
    <Component paginationType="advanced" currentPage={3} />
  )

  expect(getByText('21–30 из 100 элементов')).toBeInTheDocument()

  const pageSize = getByRole('combobox', {
    name: 'размер страницы',
  })
  fireEvent.click(pageSize)
  // Приходится искать через текст, потому что элемент селекта это не кнопка
  const pageSize_20 = await screen.findByText('20')
  fireEvent.click(pageSize_20)

  expect(await findByText('1–20 из 100 элементов')).toBeInTheDocument()
})

it('aria-label changed', () => {
  const firstPageLabel = 'search page'

  const { queryByRole } = render(
    <Component
      absoluteArrows
      paginationType="simple"
      ariaAttributes={{
        firstPageLabel,
      }}
    />
  )

  expect(queryByRole('button', { name: firstPageLabel })).toBeInTheDocument()

  // проверяем что дефолтные значения не перетираются
  expect(
    queryByRole('button', { name: 'последняя страница' })
  ).toBeInTheDocument()
})
