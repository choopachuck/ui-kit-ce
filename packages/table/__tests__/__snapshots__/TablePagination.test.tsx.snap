// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`render with correct props 1`] = `
<div
  class="root-0-2-1 end-0-2-5 md-0-2-3 dividerTop-0-2-19"
  data-testid="pagination"
>
  <div
    class="itemsWrapper-0-2-7"
  />
  <div
    class="wrapper-0-2-8"
  >
    <div
      class="pages-0-2-10"
    >
      <div
        class="pagesRange-0-2-11"
      >
        1 из 10 страниц
      </div>
    </div>
    <div
      class="buttonsWrapper-0-2-9"
    >
      <button
        aria-label="предыдущая страница"
        class="button-0-2-21 resetButton-0-2-20 disabled-0-2-22"
        type="button"
      >
        <svg
          class="disabledArrow-0-2-13"
          fill="currentColor"
          height="40"
          viewBox="0 0 40 40"
          width="40"
        >
          <rect
            fill="white"
            fill-opacity="0.01"
            height="40"
            width="40"
          />
          <path
            clip-rule="evenodd"
            d="M24 15.499L22.4 14L16 19.9995L22.4 26L24 24.4999L19.2 19.9995L24 15.499Z"
            fill="black"
            fill-rule="evenodd"
          />
        </svg>
      </button>
      <button
        aria-label="следующая страница"
        class="button-0-2-21 resetButton-0-2-20"
        type="button"
      >
        <svg
          class=""
          fill="currentColor"
          height="40"
          viewBox="0 0 40 40"
          width="40"
        >
          <rect
            fill="white"
            fill-opacity="0.01"
            height="40"
            width="40"
          />
          <path
            clip-rule="evenodd"
            d="M16 15.499L17.6 14L24 19.9995L17.6 26L16 24.4999L20.8 19.9995L16 15.499Z"
            fill="black"
            fill-rule="evenodd"
          />
        </svg>
      </button>
    </div>
  </div>
</div>
`;
